tests/kubectl_test.go

```diff
@@ -158,7 +158,7 @@ var _ = Describe("[sig-compute]oc/kubectl integration", func() {
 				tests.AddUserData(vmi, "cloud-init", "#!/bin/bash\necho 'hello'\n")
 
 				By("Starting the VirtualMachineInstance")
-				vmi = tests.RunVMIAndExpectLaunchWithIgnoreWarningArg(vmi, tests.MigrationWaitTime, false)
+				vmi = tests.RunVMIAndExpectLaunch(vmi, tests.MigrationWaitTime)
```

Comment:
This is an example of where the vmi is not being refreshed.
`RunVMIAndExpectLaunch` is not returning a refreshed VMI after the wait period.

Reply:
Thanks, I know get you: `RunVMIAndExpectLaunchWithIgnoreWarningArg` returns vmi after the wait, while `RunVMIAndExpectLaunch` returns an older vmi, right after it was created. I did not intend this change of semantics. But besides recognizing and stating it, does it matter in the context of these tests?

Reply:
I have not checked if it later actions require an updated VMI. Generally, if there is any update to the VMI in this "lost" period, a following change attempt on the VMI may fail (due to versioning or collision in data).

I think it is unsafe to leave a VMI object in a non updated state, just because someone may assume it is. (the VMI is returned, but it is not updated, which is IMO unexpected)

The main problem I see is that such things may be flaky and easy to miss. In the way the CI works today, it will just retry until it will work and we will miss such cases. Only later we may find such errors occurs too often and get to the bottom of why.

Reply:
These are all reasons to return fresh version of VMI in the end of `RunVMIAndExpectLaunch`? We use it across all tests, not just here. I this your suggestion?

Reply:
Yes.
I do not know why `RunVMIAndExpectLaunch` is not doing this already. And I do not think it should perform an explicit `Get` at the end, it should happen all the way from its inner implementation. (makes no sense to pressure the API server with endless requests)

Reply:
resolved by PR #7271, please re-review.